pkg/apiutil/serverapi/middleware.go

```diff
@@ -40,32 +41,76 @@ const (
 
 var initHTTPClientOnce sync.Once
 
+// dialClient used to dail http request.
+var dialClient = &http.Client{
+	Transport: &http.Transport{
+		DisableKeepAlives: true,
+	},
+}
+
+type runtimeServiceAuth struct {
+	s *server.Server
+}
+
+// NewRuntimeServiceAuth checks if the path is invalid.
+func NewRuntimeServiceAuth(s *server.Server) *runtimeServiceAuth {
+	return &runtimeServiceAuth{s: s}
+}
+
+func (h *runtimeServiceAuth) ServeHTTP(w http.ResponseWriter, r *http.Request, next http.HandlerFunc) {
+	if IsServiceAllowed(h.s, r.RequestURI) {
+		next(w, r)
+		return
+	}
+
+	http.Error(w, "no service", http.StatusServiceUnavailable)
+}
+
+// IsServiceAllowed checks the service through the path.
+func IsServiceAllowed(s *server.Server, path string) bool {
+	opt := s.GetServerOption()
+	cfg := opt.LoadPDServerConfig()
+	if cfg != nil {
+		for _, allow := range cfg.RuntimeServices {
+			if len(allow) != 0 && strings.Contains(path, allow) {
+				return true
+			}
+		}
+	}
+	// for core path
+	if strings.Contains(path, "api/v1") {
+		return true
+	}
+	return false
+}
+
 type redirector struct {
 	s *server.Server
 }
 
-func newRedirector(s *server.Server) *redirector {
+// NewRedirector redirects request to the leader if needs to be handled in the leader.
+func NewRedirector(s *server.Server) *redirector {
 	return &redirector{s: s}
 }
 
 func (h *redirector) ServeHTTP(w http.ResponseWriter, r *http.Request, next http.HandlerFunc) {
-	allowFollowerHandle := len(r.Header.Get(allowFollowerHandle)) > 0
+	allowFollowerHandle := len(r.Header.Get(AllowFollowerHandle)) > 0
 	if !h.s.IsClosed() && (h.s.GetMember().IsLeader() || allowFollowerHandle) {
```

Comment:
```suggestion
	if !h.s.IsClosed() && (allowFollowerHandle || h.s.GetMember().IsLeader()) {
```

Will short circuit operations be faster?